cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,71 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto const filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto constexpr num_rows = 10;
+
+  template <typename T>
+  create_random_column_wrapper(int32_t num_rows)
+  {
+    auto values = random_values<T>(num_rows);
+    return column_wrapper<T>(values.begin(), values.end());
+  }
+
+  auto int8_column     = create_random_column_wrapper<int8_t>(num_rows);
+  auto int16_column    = create_random_column_wrapper<int16_t>(num_rows);
+  auto int32_column    = create_random_column_wrapper<int32_t>(num_rows);
+  auto int64_column    = create_random_column_wrapper<int64_t>(num_rows);
+  auto uint8_column    = create_random_column_wrapper<uint8_t>(num_rows);
+  auto uint16_column   = create_random_column_wrapper<uint16_t>(num_rows);
+  auto uint32_column   = create_random_column_wrapper<uint32_t>(num_rows);
+  auto uint64_column   = create_random_column_wrapper<uint64_t>(num_rows);
+  auto float32_column  = create_random_column_wrapper<float>(num_rows);
+  auto float64_columnn = create_random_column_wrapper<double>(num_rows);
+
+  auto input_columns = std::vector<cudf::column_view>{int8_column,
+                                                      int16_column,
+                                                      int32_column,
+                                                      int64_column,
+                                                      uint8_column,
+                                                      uint16_column,
+                                                      uint32_column,
+                                                      uint64_column,
+                                                      float32_column,
+                                                      float64_column};
+
+  auto input_table = cudf::table_view{input_columns};
+
+  // write that dataframe to a csv using the default options to some temporary file
+  auto writer_options =
+    cudf_io::csv_writer_options::builder(cudf_io::sink_info{"issue.csv"}, input_table);
+  cudf_io::write_csv(writer_options);
+
+  // read the temp csv file using default options
+  auto read_options = cudf_io::csv_reader_options::builder(cudf_io::source_info{filepath})
+                        .header(-1)
```

Comment:
If I remember correctly, the issue is that the write+read should round-trip data correctly without the header parameter. In other words, writer should by default include a header, and the reader by default assumes the header is present.